### ISSUE TITLE
Very high losses when adding training samples to NER model

### ISSUE THREAD

**Issue by pythonBerg** [explosion/spaCy#2783]
I have an en model trained for a custom set of entities. Sample size >3000 items across about 10-15 entities. Results are fair to good in use...When attempting to add new annotations to this model, I am getting some alarming results...losses exceed 300k and increase each iteration. Documents can be quite large...

I am using the ner_training code found in "examples" as is with the only change being a call to db to generate training data.  Running in a linux vm, ubuntu 18.04.  Using CPU, not GPU because I cannot get GPU working through vm and windows GPU won't compile...

Please help me understand if these very high losses are expected. They are 2+ orders of magnitude larger than the ones observed during model training from blank

if not model and output_dir:
        model=output_dir

    if model is not None:
        nlp = spacy.load(model)  # load existing spaCy model
        print("Loaded model '%s'" % model)
    else:
        nlp = spacy.blank('en')  # create blank Language class
        print("Created blank 'en' model")

    if 'ner' not in nlp.pipe_names:
        ner = nlp.create_pipe('ner')
        nlp.add_pipe(ner, last=True)
    else:
        ner = nlp.get_pipe('ner')

    TRAIN_DATA = load_ner_data()
    # add labels
    for _, annotations in TRAIN_DATA:
        for ent in annotations.get('entities'):
            ner.add_label(ent[2])

    # get names of other pipes to disable them during training
    other_pipes = [pipe for pipe in nlp.pipe_names if pipe != 'ner']
    with nlp.disable_pipes(*other_pipes):  # only train NER
        optimizer = ner.create_optimizer()
        for itn in range(n_iter):
            random.shuffle(TRAIN_DATA)
            losses = {}
            for text, annotations in TRAIN_DATA:
                nlp.update(
                    [text],  # batch of texts
                    [annotations],  # batch of annotations
                    drop=0.5,  # dropout - make it harder to memorise data
                    sgd=optimizer,  # callable to update weights
                    losses=losses)
            print(losses)

**Comment by honnibal** [comment]
Hi @pythonBerg ,

If you're training with a custom set of entities, and don't have the original entities annotated in your data, then the model is learning that every instance of the previous entities it predicts is incorrect, e.g. every time it predicts PERSON, and the data doesn't have that annotated, it has to learn that was a mistake.

One solution is to annotate your new data with the previous entity types, perhaps even automatically using the initial model.

**Comment by pythonBerg** [comment]
Thanks for the clarification. Just to be clear, all I am providing is
additional instances of the entity types already established. Lets say I
have TYPE1 and TYPE2, already trained on hundreds of samples...I then pass
more TYPE1 and TYPE2. Because I am only providing docs that are fully
annotated with the established new types, how could it be determining that
previous samples of this same type are incorrect? Must all training for new
entities always contain all samples, even with an established base? That's
going to slow me down...I am hoping even more for enhanced GPU support in
future on all platforms.  Thanks

**Comment by honnibal** [comment]
Should be addressed by https://github.com/explosion/spaCy/pull/3075

See also the experimental `rehearse()` method in the new spaCy nightly: https://github.com/explosion/spaCy/pull/3035
